vaadin-notification-flow-parent/vaadin-notification-flow/src/main/java/com/vaadin/flow/component/notification/Notification.java

```diff
@@ -219,7 +222,12 @@ private void initBaseElementsAndListeners() {
     private void removeAutoAdded() {
         if (autoAddedToTheUi && !isOpened()) {
             autoAddedToTheUi = false;
-            getElement().removeFromParent();
+            try {
+                getElement().removeFromParent();
+            } catch (IllegalArgumentException e) {
+                LoggerFactory.getLogger(Notification.class).info(
+                        "Attempting to remove detached Notification. If this happended due PreserveOnRefresh, it is normal.");
```

Comment:
I'm a bit concerned about logging this. This is not a method that the user intentionally calls, and they can not really do anything about this, which I understand is more of a workaround for a Flow issue.